### ISSUE TITLE
Jest snapshot test: TypeError: Cannot read property 'monitorScrollValue' of null

### ISSUE THREAD

**Issue by dbertella** [reactjs/react-modal#254]
### Summary:
I'm trying to make a snapshot test with jest and I got this error: TypeError: Cannot read property 'monitorScrollValue' of null and the test fail without making any snapshot
### Steps to reproduce:

 1. make a snapshot test following jest tutorial, on a component with a modal in it

### Expected behavior:
Snapshot is created
### Additional notes:
I'm using create react app.

**Comment by albertolive** [comment]
I'm having the same issue with this code:

`const component = ReactTestUtils.renderIntoDocument(<InscriptionMultiChatPopup
        show={true}
        inscriptions={inscriptionsMock}
        selectedInscriptions={selectedInscriptionsMock} />),
      node = ReactTestUtils.findRenderedDOMComponentWithClass(component, 'modal');

    onClickMock = props => {
      console.log(props);
    };

    ReactTestUtils.Simulate.click(node);
    expect(clickSpy).toHaveBeenCalled();`

**Comment by diasbruno** [comment]
@dbertella @albertolive did you find a way to solve this?

**Comment by albertolive** [comment]
I ended up using Jest with enzyme.

**Comment by brandondurham** [comment]
@albertolive — Any details on what worked for you? Having trouble even finding the modal after triggering it.

**Comment by dbertella** [comment]
I'm not sure what worked for me I think I just moked the library `jest.mock('react-modal', () => 'Modal');`. If you use shallow render from enzyme you don't need to mock the library manually so I guess it's a better choice. 

I'm sorry it was an old project that I don't have access anymore but if I can help you in some other way just let me know

**Comment by indiesquidge** [comment]
@brandondurham, if you are using Enzyme, you should be able to get access to the modal with the following

```javascript
const wrapper = shallow(
  <Modal contentLabel='Example Modal' isOpen />
)
```

**Note**: the `isOpen` prop must be passed in for the modal to be opened on the page. (The `contentLabel` prop is a required prop.)

If you are attempting to access the Modal's contents or test something that happens in a lifecycle method, trying using the `mount` render function instead of `shallow`. (Check out this [comment](https://github.com/airbnb/enzyme/issues/465#issuecomment-227697726) for a better understanding of when to use each renderer.)

If you are still having issues, feel free to post more details about what you are trying to test :)

**Comment by indiesquidge** [comment]
@diasbruno, the issue of the `<ReactModal>` component rendering null is still an issue when attempting to create a snapshot test.

### Tl;dr 

_How does one create a Jest snapshot test for ReactModal?_
_The rendered output always seems to be null._

### More Info

Unfortunately, I can't create a simple JSBin or Codepen since I can't seem to figure out how to include/access Jest or `react-test-renderer` in a script tag, but I did create a very simple example using Create React App.

You can see [here](https://github.com/indiesquidge/testing-react-modal/commit/be74b97b1d3070756c750d6005781a257c5b0029#diff-6659d4de5e16b23a9f2d29064e1604e9R3) that the rendered output of the snapshot is `null`. Why is that?

```javascript
it('renders correctly (snapshot)', () => {
  const renderer = ReactTestRenderer.create(
    <ReactModal isOpen contentLabel='Example Modal' />
  )

  expect(renderer.toJSON()).toMatchSnapshot() // snapshot output is null
})
```

It doesn't seem like the output is always null, as I have no trouble testing this component with enzyme, and it _appears_ to render correctly with `ReactDOM.render`. However, this is only based on the smoke test passing, which would still work if it was returning `null`

```javascript
it('renders without crashing', () => {
  ReactDOM.render(
    React.createElement('div', {}, null),
    document.createElement('div')
  )
})
```

**Comment by indiesquidge** [comment]
The more I think about it, the more I'm realizing that a snapshot test on my end may sound like a strange thing to do.

If I was using the ReactModal component as-is, I wouldn't bother with a snapshot test. However, my use case for ReactModal may be a bit different than others in that I it is acting as somewhat of a "private component" in my codebase; I wrap a separate component

**Comment by diasbruno** [comment]
@indiesquidge Thanks for the info.

The `Modal` render method returns `null`. Maybe the component is not mounted correctly.

**Comment by indiesquidge** [comment]
> Maybe the component is not mounted correctly when the snapshot is taken.

@diasbruno, I'm not sure I understand. Isn't null the [expected return value](https://github.com/reactjs/react-modal/blob/master/src/components/Modal.js#L162)?

**Comment by diasbruno** [comment]
This one is tricky...if we are talking about the class, it's correct. If we are talking about the actual modal  tree, it is not.

Since the modal is mounted where `parentSelector={...}` returns, so you can't get a snapshot of it. 
Checkout PR #440. It should make it easy to test.

**Comment by albert-olive** [comment]
This is what I did:

    jest.mock('react-modal', () => {
      return ({ children }) => {
        return (
          <div className="modal">{children}</div>
        );
      };
    });

Also I could include one onClick and trigger it with enzyme like this:

    wrapper.find('modal').simulate('click');

**Comment by albertolive** [comment]
This is what I did:

    jest.mock('react-modal', () => {
      return ({ children }) => {
        return (
          <div className="modal">{children}</div>
        );
      };
    });

Also I could include one onClick and trigger it with enzyme like this:

    wrapper.find('modal').simulate('click');

**Comment by GunaSekhar1** [comment]
How to solve this

TypeError: Cannot read property 'create' of undefined

      at Object.<anonymous> (node_modules/react-native-vector-icons/lib/icon-button.js:29:1596)
      at Object.<anonymous> (node_modules/react-native-vector-icons/lib/create-icon-set.js:21:233)
      at Object.<anonymous> (node_modules/react-native-vector-icons/MaterialIcons.js:6:20)
